draft-ietf-tls-tls13.md

```diff
@@ -4375,6 +4375,91 @@ currently deployed and properly configured TLS implementations.
 %%### Ticket Establishment
 %%### Updating Keys
 
+# State Machine
+
+This section provides a summary of the legal state transitions for the
+client and server handshakes.  State names (in all capitals, e.g.,
+START) have no formal meaning but are provided for ease of
+comprehension.  Messages which are sent only sometimes are indicated
+in [].
+
+## Client
+
+~~~~
+                  +-----> START
+             Recv |        |
+            Hello |        | Send ClientHello
+            Retry |        v
+        +-> Request +---- WAIT_SH
+    Can |                    | Recv ServerHello
+   Send |                    V
+  Early |                 WAIT_EE
+   Data |                    | Recv EncryptedExtensions
+        |           +--------+--------+
+        |     Using |                 | Using Certificate
+        |       PSK |                 v
+        |           |            WAIT_CERT_CR
+        |           |        Recv |       | Recv CertificateRequest
+        |           | Certificate |       v
+        |           |             |    WAIT_CERT
+        |           |             |       | Recv Certificate
+        |           |             v       v
+        |           |              WAIT_CV
+        |           |                 | Recv CertificateVerify
+        |           +> WAIT_FINISHED <+
+        |                  | Recv Finished
+        |                  |
+        +->                | [Send EndOfEarlyData]
+                           | [Send Certificate [+ CertificateVerify]]
+                           | Send Finished
+ Can Send                  v
+ App Data -->          CONNECTED
+ After
+ Here
+~~~~
+
+## Server
+
+~~~~
+                      +----> START
+                 Send |        |
+                Hello |        | Recv ClientHello
+                Retry |        |
+              Request |        v
+                      +--- RECVD_CH
+                               | Select parameters
```

Comment:
Capitalization is also a bit inconsistent. For the most part, Everything Is Written In Title Case:
- Can Send Early Data
- Can Send App Data After Here
- Client Auth
- No Auth
- Using Certificate
- Recv Early Data
- Recv Empty Certificate

But "Select parameters" is sentence-case. FWIW, I think the Excessive Title Case Feels A Little Weird and might suggest changing the others instead. (Though keep messages in UpperCamelCase of course.) These feel more like sentences.